frontend/src/metabase/common/components/QueryColumnPicker/QueryColumnPicker.tsx

```diff
@@ -7,22 +7,30 @@ import { singularize } from "metabase/lib/formatting";
 
 import * as Lib from "metabase-lib";
 
+import {
+  BinningStrategyPickerPopover,
+  TemporalBucketPickerPopover,
+} from "./BucketPickerPopover";
+
 const DEFAULT_MAX_HEIGHT = 610;
 
-interface QueryColumnPickerProps {
+type ColumnListItem = Lib.ColumnDisplayInfo & {
+  column: Lib.ColumnMetadata;
+};
+
+export interface QueryColumnPickerProps {
   className?: string;
   query: Lib.Query;
   stageIndex: number;
   columnGroups: Lib.ColumnGroup[];
+  hasBinning?: boolean;
+  hasTemporalBucketing?: boolean;
   maxHeight?: number;
+  checkIsColumnSelected: (item: ColumnListItem) => boolean;
```

Comment:
`QyeryColumnPicker` users have to implement the `isSelected` check because so far there's no shared "selected" column state:

* selected `orderableColumns` have `orderByPosition: <int>` property in their display info
* selected `breakoutableColumns` have `breakoutPositon: <int>` property in their display info

Check out `BreakoutStep` and `SortStep` components for reference implementations